tuxemon/core/components/player.py

```diff
@@ -698,6 +698,39 @@ def get_adjacent_tiles(self, curr_loc, game):
             adj_tiles.append((curr_loc[0]+1,curr_loc[1]))
         return adj_tiles
 
+    def add_item(self, action):
+        """Adds an item to the current player's inventory. The action parameter must contain an
+        item name to look up in the item database.
+
+        :param game: The main game object that contains all the game's variables.
+        :param action: The action (tuple) retrieved from the database that contains the action's
+            parameters
+
+        :type game: core.control.Control
+        :type action: Tuple
+
+        :rtype: None
+        :returns: None
+
+        **Example:**
+
+        >>> action.__dict__
+        {
+            "type": "add_item",
+            "parameters": [
+                "Potion"
+            ]
+        }
+
+        """
+        item_to_add = item.Item(action.parameters[0])
+
+        # If the item already exists in the player's inventory, add to its quantity, otherwise
+        # just add the item.
+        if item_to_add.name in [i.name for i in self.inventory.keys()]:

```

Comment:
Hi,

This PR looks good, but I had a question about the `add_item` logic that you used here.

Was there any reason you used a list comprehension here? To check for a key in a dictionary in python you can simply use `if item_to_add.name in self.inventory`. Using a list comprehension adds a little unnecessary complexity.